src/dsp_tools/commands/excel2json/lists_compliance_checks.py

```diff
@@ -100,7 +162,138 @@ def make_col_names(lang: str) -> set[str]:
     if missing_cols := expected_cols - set(cols):
         return {
             "missing translations": f"All nodes must be translated into the same languages. "
-            f"One or more translations for the following column(s) are missing: "
+            f"Based on the languages used, the following column(s) are missing: "
             f"{', '.join(missing_cols)}"
         }
     return {}
+
+
+def _make_all_list_content_compliance_checks(
+    excel_dfs: dict[str, dict[str, pd.DataFrame]],
+) -> None:
+    """
+    This function checks if the content of the excel files is compliant with the expected format.
+
+    Args:
+        excel_dfs: dictionary with the excel file name as key
+                    and a dictionary with the sheet name as key and the dataframe.
+
+    Raises:
+        InputError: If any node is missing translations
+    """
+    _check_all_excels_if_any_nodes_miss_translations(excel_dfs)
+    _check_all_excel_for_erroneous_entries(excel_dfs)
+
+
+def _check_all_excels_if_any_nodes_miss_translations(excel_dfs: dict[str, dict[str, pd.DataFrame]]) -> None:
+    problems = []
+    for filename, excel_sheets in excel_dfs.items():
+        missing_translations: list[Problem] = [
+            p
+            for sheet_name, df in excel_sheets.items()
+            if (p := _check_sheet_if_any_nodes_miss_translations(df, sheet_name)) is not None
+        ]
+        if missing_translations:
+            problems.append(ListExcelProblem(filename, missing_translations))
+    if problems:
+        msg = ListCreationProblem(problems).execute_error_protocol()
+        logger.error(msg)
+        raise InputError(msg)
+
+
+def _check_sheet_if_any_nodes_miss_translations(
+    df: pd.DataFrame, sheet_name: str
+) -> MissingTranslationsSheetProblem | None:
+    col_endings = [str(num) for num in _get_hierarchy_nums(df.columns)]
+    col_endings.append("list")
+    languages = _get_all_languages_for_columns(df.columns)
+    all_cols = _make_columns(col_endings, languages)
+    problems = []
+    for column_group in all_cols:
+        problems.extend(_check_one_hierarchy(column_group, df))
+    if problems:
+        return MissingTranslationsSheetProblem(sheet_name, problems)
+    return None
+
+
+def _check_one_hierarchy(columns: list[str], df: pd.DataFrame) -> list[MissingNodeTranslationProblem]:
+    problems = []
+    for i, row in df.iterrows():
+        if problem := _check_one_node_for_translations(row, columns, int(str(i))):
+            problems.append(problem)
+    return problems
+
+
+def _make_columns(nums: list[str], languages: set[str]) -> list[list[str]]:
+    return [[f"{lang}_{num}" for lang in languages] for num in nums]
```

Comment:
I don't actually want a flat list additionally something with a range or number of repetitions is also not what I want because if they deleted all columns _2 but have, 1, 3 then it is still valid input.